app/services/access_token_verifier.rb

```diff
@@ -0,0 +1,53 @@
+class AccessTokenVerifier
+  include ActionView::Helpers::TranslationHelper
+  include ActiveModel::Model
+
+  validate :validate_access_token
+
+  def initialize(http_authorization_header)
+    @http_authorization_header = http_authorization_header
+    @identity = nil
+  end
+
+  def submit
+    FormResponse.new(success: valid?, errors: errors)
+  end
+
+  def identity
+    valid? ? @identity : nil
+  end
+
+  private
+
+  attr_reader :http_authorization_header
+
+  def validate_access_token
+    access_token = extract_access_token(http_authorization_header)
+    load_identity(access_token) if access_token
+  end
+
+  def load_identity(access_token)
+    identity = Identity.where(access_token: access_token).first
+
+    if identity && Pii::SessionStore.new(identity.session_uuid).ttl.positive?
+      @identity = identity
+    else
+      errors.add(:access_token, t('openid_connect.user_info.errors.not_found'))
+    end
+  end
+
+  def extract_access_token(header)
+    if header.blank?
+      errors.add(:access_token, t('openid_connect.user_info.errors.no_authorization'))
+      return
+    end
+
+    bearer, access_token = header.split(' ', 2)
```

Comment:
nit: should this be a pattern `/\ +/` in case the incoming request has more than one space separating?